integration_tests/src/main/python/orc_test.py

```diff
@@ -663,3 +663,7 @@ def test_read_type_casting_integral(spark_tmp_path, offset, reader_confs, v1_ena
     assert_gpu_and_cpu_are_equal_collect(
         lambda spark: spark.read.schema(rs).orc(data_path),
         conf=all_confs)
+
+def test_orc_read_varchar_as_string(std_input_path):
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: spark.read.schema("id bigint, name string").orc(std_input_path + "/test_orc_varchar.orc"))
```

Comment:
it would be more obvious what is being tested if the file was created on the fly in this test. It would also avoid checking in yet another binary resource.

Reply:
AFAICT Spark is unable to create varchar files, as the type is always upgraded to a string with a warning.  That's why I checked in a pre-built file built with Hive.  I don't think there's any way to access Hive direct from pyspark (without hacking through the raw JVM interface) and Hive isn't required to be present with the Spark distribution.

If there's a way to create this file directly in the test, happy to update the test accordingly.

Reply:
I poked around Spark and confirm. I did not realize it's difficult